openapi-cli/src/test/java/io/ballerina/openapi/generators/schema/ConstraintTests.java

```diff
@@ -103,6 +103,15 @@ public void testRecordFiledConstraintWithZeroValue() throws IOException, Balleri
         List<Diagnostic> diagnostics = getDiagnostics(syntaxTree);
         Assert.assertTrue(diagnostics.isEmpty());
     }
+    /**
+     * example warnings for union type has constraint
+     *
+     * WARNING: `@constraint` support will not be available with union type in given `service_class` field
+     * WARNING: `@constraint` support will not be available with union type in given `tax_rates` field
+     * WARNING: `@constraint` support will not be available with union type in given `tax_rates_anyOf` field
+     * WARNING: `@constraint` support will not be available with union type in given `tax_rates_oneOF_array` field
+     * WARNING: `@constraint` support will not be available with union type in given `tax_rates_anyOf_array` field
+     */
```

Comment:
Let's add negative test cases to ensure that all the above unsupported scenarios are handled by the tool.

Reply:
I have added with the same test, will separate the test with warnings  :)